plugin/storage/factory_test.go

```diff
@@ -392,29 +401,69 @@ func TestPublishOpts(t *testing.T) {
 	})
 }
 
-type errorCloseFactory struct {
-	closeErr error
+func TestCreateLockAndSamplingStore(t *testing.T) {
+	supports := &errorFactory{}
+	doesNotSupport := &errorFactory{
+		lockAndSamplingStore: storage.ErrLockAndSamplingStoreNotSupported,
+	}
+	breaks := &errorFactory{
+		lockAndSamplingStore: errors.New("wups"),
+	}
+
+	// test succeeds (supported or not)
+	f := Factory{
+		factories: map[string]storage.Factory{
+			"supports":         supports,
+			"does-not-support": doesNotSupport,
+		},
+	}
+
+	_, _, err := f.CreateLockAndSamplingStore()
```

Comment:
Can you assert that it returns `supports`?

Reply:
Added logic to test that the appropriate mocks were returned.